be/src/runtime/data_stream_recvr.cc

```diff
@@ -205,6 +216,42 @@ bool DataStreamRecvr::SenderQueue::is_finished() const {
     return _is_cancelled || (_num_remaining_senders == 0 && _chunk_queue.empty());
 }
 
+bool DataStreamRecvr::SenderQueue::has_chunk() {
+    std::unique_lock<std::mutex> l(_lock);
+    if (_is_cancelled) {
+        return true;
+    }
+
+    if (_chunk_queue.empty() && _num_remaining_senders > 0) {
+        return false;
+    }
+
+    return true;
+}
+
+bool DataStreamRecvr::SenderQueue::try_get_chunk(vectorized::Chunk** chunk) {
```

Comment:
Why get chunk successfully return  false？ failed  return true ?

Reply:
Because in pipeline, try_get_chunk return true mean there is no datas or query in cancelled, false mean get chunk. there is no other cases.

Reply:
So I think your return true or false is reverse.